metric_learn/lmnn.py

```diff
@@ -43,13 +46,17 @@ def __init__(self, k=3, min_iter=50, max_iter=1000, learn_rate=1e-7,
     self.convergence_tol = convergence_tol
     self.use_pca = use_pca
     self.verbose = verbose
+    super(_base_LMNN, self).__init__(preprocessor)
 
 
 # slower Python version
 class python_LMNN(_base_LMNN):
 
   def _process_inputs(self, X, labels):
```

Comment:
You mean for the python_LMNN inputs ? Like the specific error messages "Must have one label per point" or 'use_pca does nothing for the python_LMNN implementation' ? I agree that the name `_process_inputs` here does not tell the difference from `_prepare_inupts` explicitly

Reply:
I guess I was a bit confused, but maybe add a comment in the function to point out that some checks are done to avoid that parameters from the Shogun version are passed to the Python version

In the near future we should improve the performance of our own implementation so that we can remove this dependency to Shogun

Reply:
I agree, will do
I'll raise an issue for removing the Shogun dependency